src/record/partially_serialized.rs

```diff
@@ -87,25 +103,4 @@ impl PartiallySerializedRecord {
 
         Ok((buf, checksum))
     }
-
-    async fn write_single_pass(buf: Bytes, file: &File) -> Result<u64> {
-        let len = buf.len() as u64;
-        Self::process_file_result(file.write_append_all(buf).await).map(|_| len)
-    }
-
-    async fn write_double_pass(head: Bytes, data: Bytes, file: &File) -> Result<u64> {
-        let len = (head.len() + data.len()) as u64;
-        Self::process_file_result(file.write_append_all_buffers(head, data).await).map(|_| len)
-    }
-
-    fn process_file_result(result: std::result::Result<(), std::io::Error>) -> Result<()> {
```

Comment:
You lost this result conversion! It's important to keep it because there is logic up the call chain that depends on this processing